santiment/parity-monitor
# Add comparing with Etherscan last #block

* Compare with Etherscan latest block number
  * Update Dockerfile, removed unused code/logic
  * Update docker-compose and sh scripts for local dev without node on
    host machine
  * Add Eslint & Prettier for better dev practice
  * Removed unused jslint
  * Add .env flow
 --
My proposal is using Etherscan.io API as one more validation healthy check step.
Idea to compare our Node's latest block number with Etherscan.

One more thing I've found unused code in Dockerfile and old deps like web3 library and fixed it.
I've noticed not consistent usage of code style like "semi" vs "no-semi" and fixed it with added prettier.
Also JSLint wasn't implemented right, It doesn't work and probably never worked. I fixed it and added ESlint instead of JSLint.
To run dev env you should use `sh bin/run.sh` as before
To run linter & prettier you should use `sh bin/test.sh`
I didn't add prettier and lint as pre-commit hook, only cause it's a pretty simple repository and don't spend much time on it.

P.S.
To use Etherscan API we have to setup API key. It's possible to do via env variables or .env file.
By default Etherscan checker step is off to avoid any problem with old version of this application.
Devops changes is required for proper works on production/stage.
